monkey/monkey_island/cc/resources/agents.py

```diff
@@ -14,8 +15,12 @@
 class Agents(AbstractResource):
     urls = ["/api/agents"]
 
-    def __init__(self, island_event_queue: IIslandEventQueue):
+    def __init__(self, island_event_queue: IIslandEventQueue, agent_repository: IAgentRepository):
         self._island_event_queue = island_event_queue
+        self._agent_repository = agent_repository
+
+    def get(self):
+        return self._agent_repository.get_agents(), HTTPStatus.OK
```

Comment:
How should the resource handle an error from `get_agents()`?

Reply:
`get_agents()` only raises `RetrievalError`, which means something unexpected happened. In HTTP-Land, that equates to `500: Internal Server Error`. Flask catches uncaught exceptions and returns a 500 error code.

Reply:
Oh, good to know!